src/surveys/components/DropdownResponse.tsx

```diff
@@ -0,0 +1,192 @@
+  label,
+  description,
+  question,
+  theme,
+  onAnswered,
+  submitText,
+}: DropdownProps): VNode {
+  const [selectedOption, setSelectedOption] = useState<QuestionOption | null>(
+    null
+  );
+  const [isOpen, setIsOpen] = useState(false);
+  const [filteredOptions, setFilteredOptions] = useState<QuestionOption[]>(
+    question.options as QuestionOption[]
+  );
+  const [positionBelow, setPositionBelow] = useState(true);
+  const [dropdownWidth, setDropdownWidth] = useState(100);
+  const [inputRectLeft, setInputRectLeft] = useState(0);
+  const [inputRectBottom, setInputRectBottom] = useState(0);
+  const [inputRectTop, setInputRectTop] = useState(0);
+
+  const inputRef = useRef<HTMLInputElement | null>(null);
+
+  const questionColor = theme?.question ?? '#050505';
+  const dropDownMaxHeight = 200;
+
+  const handleInputChange = (
+    event: h.JSX.TargetedEvent<HTMLInputElement, Event>
+  ) => {
+    const query = event.currentTarget.value.toLowerCase();
+    const filtered = (question.options ?? []).filter((option) =>
+      option.label.toLowerCase().includes(query)
+    );
+    setFilteredOptions(filtered);
+    setIsOpen(true);
+    setSelectedOption(null);
+  };
+
+  const handleOptionClick = (option: QuestionOption) => {
+    setSelectedOption(option);
+    setIsOpen(false);
+  };
+
+  const toggleDropdown = () => {
+    setIsOpen(!isOpen);
+  };
+
+  const calculatePosition = () => {
+    if (!inputRef.current) return;
+    const inputRect = inputRef.current.getBoundingClientRect();
+    const windowHeight = window.innerHeight;
+
+    setDropdownWidth(inputRef.current.clientWidth);
+    setInputRectLeft(inputRect.left);
+    setInputRectTop(windowHeight - inputRect.top);
+    setInputRectBottom(inputRect.bottom);
+
+    if (
+      windowHeight - inputRect.bottom < dropDownMaxHeight &&
+      inputRect.top > dropDownMaxHeight
+    ) {
+      setPositionBelow(false);
+    } else {
+      setPositionBelow(true);
+    }
+  };
+
+  useEffect(() => {
+    calculatePosition();
+  }, [inputRef.current?.clientWidth]);
+
+  useEffect(() => {
+    window.addEventListener('scroll', calculatePosition);
+    window.addEventListener('resize', calculatePosition);
+    return () => {
+      window.removeEventListener('scroll', calculatePosition);
+      window.removeEventListener('resize', calculatePosition);
+    };
+  }, []);
+
+  const dropDownPositionY: h.JSX.CSSProperties = positionBelow
+    ? { top: `${inputRectBottom}px` }
+    : { bottom: `${inputRectTop}px` };
+
+  const dropdownStyle: h.JSX.CSSProperties = {
+    width: `${dropdownWidth}px`,
+    maxHeight: `${dropDownMaxHeight}px`,
+    backgroundColor: theme?.background ?? '#FFFFFF',
+    position: 'absolute',
+    overflowY: 'auto',
+    left: `${inputRectLeft}px`,
+    ...dropDownPositionY,
+  };
+
+  const dropdownChildStyle: h.JSX.CSSProperties = {
+    padding: '8px',
+    color: questionColor,
+    backgroundColor: hexToRgba(questionColor, 0.1),
```

Comment:
@AfeezGL I think the backgroundColor for the dropdown list should still be the `theme.background`